exasol_advanced_analytics_framework/lua/src/event_loop.lua

```diff
@@ -0,0 +1,58 @@
+---
+-- @module event_loop
+--
+-- This module processes only the state transitions by executing queries returned by the Event Handler
+--
+
+
+
+local M = {}
+local exa_error = require("exaerror")
+
+_G.global_env = {
+    pquery = pquery,
+    error = error
+}
+
+---
+-- Executes the given set of queries.
+--
+function M._run_queries(queries, from_index)
+    local all_success = true
+    for i=from_index, #queries do
+        local success, result = _G.global_env.pquery(queries[i][1])
+        all_success = all_success and success
+        if not success then
+            local error_obj = exa_error.create(
+                    "E-AAF-3",
+                    "Error occurred in executing queries: " .. result.error_message)
+            _G.global_env.error(tostring(error_obj))
+        end
+    end
+    return all_success
+end
+
+---
+-- Initiates the Event Loop that handles state transition
+--
+function M.init(query_to_event_handler)
+    local status = "started"
+
+    repeat
+        local success, result = _G.global_env.pquery(query_to_event_handler)
+        if not success then
+            local error_obj = exa_error.create(
+                    "E-AAF-2",
+                    "Error occurred in calling Event Handler: " .. result.error_message)
+            _G.global_env.error(tostring(error_obj))
+        end
+        query_to_event_handler = result[1][1]
```

Comment:
So, I guess, this defines that the first result line of the event handler returns the return query.

Reply:
exactly as I wrote in the comment above.